### PR TITLE
dnsviz 0.6.5 (new formula)

### PR DESCRIPTION
**Pull request by cdeccio**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by woodruffw** [comment]
Looks like your test block failed:

```
There was an error parsing the json input: "test.json"
Error: dnsviz: failed
```

I couldn't replicate a parsing failure with `test.json` and `jq`, so I'm not sure what to make of this.

File: Formula/dnsviz.rb
Comment (MikeMcQuaid) [inline Formula/dnsviz.rb]: Indent by two characters, split this into multiple lines and don't escape the `"`. I'd suggest using `jq` to pretty-print the output for here.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Comment by cdeccio** [comment]
The issue with the json blob was that I had originally written is that it was a string--hence the escaped quotes. Then I changed it a here string, but never unquoted.  I'm not sure why it didn't get tested like that.

Anyway, I believe I've fixed the issue, but I might have made a mess of the branch and pull request.  If so, let me know, and I'll clean it up.

**Comment by MikeMcQuaid** [comment]
Squash commits by running `brew update; git checkout $YOUR_BRANCH; git rebase --interactive origin/master` and then mark each commit other than the first as `squash` or `fixup` then `git push -f`. Thanks!

**Comment by MikeMcQuaid** [comment]
@cdeccio As mentioned earlier: indent the JSON by two characters, split this into multiple lines and don't escape the ". I'd suggest using `jq` to pretty-print the output for here.

**Comment by cdeccio** [comment]
@MikeMcQuaid the json used for the test is very sparse, which is why I left it in compact form, but if you still prefer, I'll expand it out.

**Comment by MikeMcQuaid** [comment]
> @MikeMcQuaid the json used for the test is very sparse, which is why I left it in compact form, but if you still prefer, I'll expand it out.

@cdeccio It's extremely long so: yes, please.

**Comment by cdeccio** [comment]
okay, after some pain and oops, I think I've gotten it recommitted properly.

**Comment by apjanke** [comment]
Build [failed](https://bot.brew.sh/job/Homebrew%20Core/11846/) on Yosemite only.

```
 creating build/temp.macosx-10.10-intel-2.7/SWIG
    clang -fno-strict-aliasing -fno-common -dynamic -arch x86_64 -arch i386 -g -Os -pipe -fno-common -fno-strict-aliasing -fwrapv -DENABLE_DTRACE -DMACOSX -DNDEBUG -Wall -Wstrict-prototypes -Wshorten-64-to-32 -DNDEBUG -g -fwrapv -Os -Wall -Wstrict-prototypes -DENABLE_DTRACE -arch x86_64 -arch i386 -pipe -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/System/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 -I/usr/include -I/private/tmp/pip-wl5cJm-build/SWIG -c SWIG/_m2crypto_wrap.c -o build/temp.macosx-10.10-intel-2.7/SWIG/_m2crypto_wrap.o -DTHREADING
    SWIG/_m2crypto_wrap.c:5005:9: error: invalid argument type 'void' to unary expression
        if (!HMAC_Init(ctx, kbuf, klen, md)) {
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    SWIG/_m2crypto_wrap.c:5020:9: error: invalid argument type 'void' to unary expression
        if (!HMAC_Update(ctx, buf, len)) {
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~
    SWIG/_m2crypto_wrap.c:5038:9: error: invalid argument type 'void' to unary expression
        if (!HMAC_Final(ctx, blob, (unsigned int *)&blen)) {
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    SWIG/_m2crypto_wrap.c:8774:3: error: unknown type name '_STACK'; did you mean 'STACK'?
      _STACK *arg1 = (_STACK *) 0 ;
      ^~~~~~
      STACK
    /usr/include/openssl/stack.h:76:4: note: 'STACK' declared here
            } STACK;
              ^
    SWIG/_m2crypto_wrap.c:8774:27: error: expected expression
      _STACK *arg1 = (_STACK *) 0 ;
                              ^
    SWIG/_m2crypto_wrap.c:8774:19: error: use of undeclared identifier '_STACK'
      _STACK *arg1 = (_STACK *) 0 ;
                      ^
    SWIG/_m2crypto_wrap.c:8785:19: error: expected expression
      arg1 = (_STACK *)(argp1);
                      ^
    SWIG/_m2crypto_wrap.c:8785:11: error: use of undeclared identifier '_STACK'
      arg1 = (_STACK *)(argp1);
              ^
    SWIG/_m2crypto_wrap.c:8796:3: error: unknown type name '_STACK'; did you mean 'STACK'?
      _STACK *arg1 = (_STACK *) 0 ;
      ^~~~~~
      STACK
    /usr/include/openssl/stack.h:76:4: note: 'STACK' declared here
            } STACK;
              ^
    SWIG/_m2crypto_wrap.c:8796:27: error: expected expression
      _STACK *arg1 = (_STACK *) 0 ;
                              ^
    SWIG/_m2crypto_wrap.c:8796:19: error: use of undeclared identifier '_STACK'
      _STACK *arg1 = (_STACK *) 0 ;
                      ^
    SWIG/_m2crypto_wrap.c:8811:19: error: expected expression
      arg1 = (_STACK *)(argp1);
                      ^
    SWIG/_m2crypto_wrap.c:8811:11: error: use of undeclared identifier '_STACK'
      arg1 = (_STACK *)(argp1);
              ^
    SWIG/_m2crypto_wrap.c:8827:3: error: unknown type name '_STACK'; did you mean 'STACK'?
      _STACK *arg1 = (_STACK *) 0 ;
      ^~~~~~
      STACK
    /usr/include/openssl/stack.h:76:4: note: 'STACK' declared here
            } STACK;
              ^
    SWIG/_m2crypto_wrap.c:8827:27: error: expected expression
      _STACK *arg1 = (_STACK *) 0 ;
                              ^
    SWIG/_m2crypto_wrap.c:8827:19: error: use of undeclared identifier '_STACK'
      _STACK *arg1 = (_STACK *) 0 ;
                      ^
    SWIG/_m2crypto_wrap.c:8845:19: error: expected expression
      arg1 = (_STACK *)(argp1);
                      ^
    SWIG/_m2crypto_wrap.c:8845:11: error: use of undeclared identifier '_STACK'
      arg1 = (_STACK *)(argp1);
              ^
    SWIG/_m2crypto_wrap.c:8867:3: error: unknown type name '_STACK'; did you mean 'STACK'?
      _STACK *result = 0 ;
      ^~~~~~
      STACK
    /usr/include/openssl/stack.h:76:4: note: 'STACK' declared here
            } STACK;
              ^
    fatal error: too many errors emitted, stopping now [-ferror-limit=]
    20 errors generated.
    error: command 'clang' failed with exit status 1
    Running setup.py install for M2Crypto: finished with status 'error'
Cleaning up...
```

Was this failing before?

Based on a bit of googling for "openssl '_STACK' m2crypto", it looks like this may be a compatibility problem with the older versions of the system OpenSSL shipped with Yosemite and prior.

https://gitlab.com/m2crypto/m2crypto/issues/63

Based on the M2Crypto maintainer's comments there, I'd say that M2Crypto is not supported for OS X versions prior to El Capitan. You could change this formula to require El Capitan or later. 

Another option is changing it to build M2Crypto against the Homebrew-supplied OpenSSL. The comments on that issue include a workaround for doing so. The existing `onepass`, `osc`, and `saltstack` formulae also patch M2Crypto to build against the brewed OpenSSL, and you could follow their examples.

Since we try to always use the brewed openssl, that seems like the appropriate choice here.

**Comment by apjanke** [comment]
Your commit message is really long.

```
 # This is a combination of 58 commits.

# The first commit's message is:
# This is a combination of 9 commits.
# The first commit's message is:
# This is a combination of 2 commits.
# The first commit's message is:

dnsviz 0.6.5 (new formula)

# The 2nd commit message will be skipped:

#	dnsviz 0.6.5 (new formula)
...
```

Try doing a `git rebase -i master` and choose "reword" to clean it up.

**Comment by cdeccio** [comment]
> Your commit message is really long.

Yup.  I thought I had cleaned that up.  I've done it now.  Thanks for the suggestion.

**Comment by apjanke** [comment]
> Yup. I thought I had cleaned that up. I've done it now.

Commit message looks good now; thanks!

**Comment by MikeMcQuaid** [comment]
> Since we try to always use the brewed openssl, that seems like the appropriate choice here.

Yep, that's the best thing to do 👍

**Comment by cdeccio** [comment]
> > Since we try to always use the brewed openssl, that seems like the appropriate choice here.

> Yep, that's the best thing to do 👍

Okay, I'll do that.  If you have an example of another package that does that, I'll model that.  Otherwise, I'll poke around and find one myself.

**Comment by MikeMcQuaid** [comment]
@tdsmith may be able to help point you in the right direction.

**Comment by tdsmith** [comment]
It's hard to be sure what's happening here without the superenv logs; i'm guessing that the active developer path is in /Applications/Xcode.app, M2Crypto's setup.py is (naughtily) adding -I/usr/include to its command lines and it isn't being filtered out by the superenv shims, and our Homebrew paths are being added later in the invocation and getting masked by the system header.

A couple ways around this:

1. m2crypto's `setup.py build_ext` takes a `--openssl` argument https://gitlab.com/m2crypto/m2crypto/blob/master/setup.py#L85. Maybe most straightforward thing to do would be to install m2crypto by manually invoking setup.py.

2. we can patch m2crypto's setup.py to stop adding "system default" paths to the command line

3. we can inreplace our paths into m2crypto's setup.py

Option 1 looks like:

```ruby
venv = virtualenv_create(libexec)
resource("m2crypto").stage do
  system libexec/"bin/python", "setup.py", "build_ext", "--openssl=#{Formula["openssl"].opt_prefix}", "install"
end
res = resources.map(&:name).to_set - ["m2crypto"]
res.each do |r|
  venv.pip_install resource(r)
end
venv.pip_install_and_link buildpath
```

**Comment by cdeccio** [comment]
@tdsmith Thanks for the suggestions.  I've implemented your suggested option 1 and recommitted.

**Comment by woodruffw** [comment]
LGTM, will wait for a second :+1: from either @tdsmith or @MikeMcQuaid before merging.

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Comment by cdeccio** [comment]
@MikeMcQuaid Thank you for helping me through the process and for accepting the PR.  Glad to be a contributor.

**Pull request closed without merging**
